src/main/java/rx/internal/operators/OnSubscribeSingle.java

```diff
@@ -0,0 +1,72 @@
+package rx.internal.operators;
+
+import rx.Observable;
+import rx.Single;
+import rx.SingleSubscriber;
+import rx.Subscriber;
+
+import java.util.NoSuchElementException;
+
+/**
+ * Allows conversion of an Observable to a Single ensuring that exactly one item is emitted - no more and no less.
+ * Also forwards errors as appropriate.
+ */
+public class OnSubscribeSingle<T> implements Single.OnSubscribe<T> {
+
+    private final Observable<T> observable;
+
+    public OnSubscribeSingle(Observable<T> observable) {
+        this.observable = observable;
+    }
+
+    @Override
+    public void call(final SingleSubscriber<? super T> singleSubscriber) {
+        observable.subscribe(new Subscriber<T>() {
```

Comment:
Thanks that helps quite a bit in understanding. However, unlike in your post, I cannot pass the child subscriber to the parent since the child has the SingleSubscriber type which isn't accepted by the Subscriber constructor.